akka-actor-typed-tests/src/test/java/jdocs/akka/typed/FSMDocTest.java

```diff
@@ -0,0 +1,184 @@
+    //#simple-events
+
+    //#simple-events
+    public enum Flush implements Event {
+        Flush
+    }
+    //#simple-events
+
+    static
+    //#simple-events
+    public final class Queue implements Event {
+        private final Object obj;
+
+        public Queue(Object obj) {
+            this.obj = obj;
+        }
+
+        public Object getObj() {
+            return obj;
+        }
+    }
+
+    //#simple-events
+    static public final class Batch {
+        private final List<Object> list;
+
+        public Batch(List<Object> list) {
+            this.list = list;
+        }
+
+        public List<Object> getList() {
+            return list;
+        }
+        //#boilerplate
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            Batch batch = (Batch) o;
+
+            return list.equals(batch.list);
+        }
+
+        @Override
+        public int hashCode() {
+            return list.hashCode();
+        }
+
+        @Override
+        public String toString() {
+            final StringBuilder builder = new StringBuilder();
+            builder.append("Batch{list=");
+            list.stream().forEachOrdered(e -> {
+                builder.append(e);
+                builder.append(",");
+            });
+            int len = builder.length();
+            builder.replace(len, len, "}");
+            return builder.toString();
+        }
+        //#boilerplate
+    }
+
+    //#storing-state
+    interface Data {
+    }
+    //#storing-state
+
+    static
+    //#storing-state
+    final class Todo implements Data {
+        private final ActorRef<Batch> target;
+        private final List<Object> queue;
+
+        public Todo(ActorRef<Batch> target, List<Object> queue) {
+            this.target = target;
+            this.queue = queue;
+        }
+
+        public ActorRef<Batch> getTarget() {
+            return target;
+        }
+
+        public List<Object> getQueue() {
+            return queue;
+        }
+        //#boilerplate
+
+        @Override
+        public String toString() {
+            return "Todo{" +
+                    "target=" + target +
+                    ", queue=" + queue +
+                    '}';
+        }
+
+        public Todo addElement(Object element) {
+            List<Object> nQueue = new LinkedList<>(queue);
+            nQueue.add(element);
+            return new Todo(this.target, nQueue);
+        }
+
+        public Todo copy(List<Object> queue) {
+            return new Todo(this.target, queue);
+        }
+
+        public Todo copy(ActorRef<Batch> target) {
+            return new Todo(target, this.queue);
+        }
+        //#boilerplate
+    }
+    //#storing-state
+
+    //#simple-state
+    // FSM states represented as behaviors
+    private static Behavior<Event> uninitialized() {
+        return Behaviors.receive(Event.class)
```

Comment:
even though this is not persistent I wonder if we should avoid `Event` here, it's still more like a command or a message?

Reply:
I see the point of using Event since untyped FSM is using it and the intention of this documentation is like a "migration guide"

Reply:
Yep wanted to stick with the naming from untyped fsm